flake.nix

```diff
@@ -0,0 +1,164 @@
+{
+  description = "The EGIL SCIM client";
+
+  inputs = {
+    nixpkgs.url = "nixpkgs/nixos-21.05";
+    flake-utils.url = "github:numtide/flake-utils";
+    debug-info-utils.url = "git+https://git.sr.ht/~ilkecan/debug-info-utils";
+    overlay-utils.url = "git+https://git.sr.ht/~ilkecan/overlay-utils";
+    source-utils.url = "git+https://git.sr.ht/~ilkecan/source-utils";
+    version-utils.url = "git+https://git.sr.ht/~ilkecan/version-utils";
```

Comment:
We should flake these up or atleast use proper git references. Are these dependencies by the same author?

Reply:
I am the author of these repositories. I realized I reused these functions so I created separate repositories. Not sure about what you mean by proper git references. Used revisions are already locked in with the lock file.

Reply:
oh, sorry. Initially, I thought that these are sources/dependencies that are part of this package and did not check what they actually are.

Do you think maybe we could upstream them to nixpkgs? I think that would be nice.

Reply:
Also, I doubt that author is going to be too fond of such tiny dependencies :thinking:

Reply:
I am not sure if they are really relevant or worthy to be upstreamed to nixpkgs or flake-utils. But I didn't want to repeat myself across different flakes.

Reply:
I decided to create an umbrella repository for these snippets as the first step. After that I will try to upstream them to a bigger project if that makes sense.

Reply:
Merged them to a single repository for now.
Related: 4e3b9fb4d3ab4931592339a7ba0e8e8fd016f90d